src/utils/aotf.js

```diff
@@ -0,0 +1,586 @@
+mutationStatus[TaskState.SUCCEEDED] = TaskState.SUCCEEDED
+mutationStatus[TaskState.FAILED] = TaskState.FAILED
+mutationStatus[TaskState.SUBMIT_FAILED] = TaskState.SUBMIT_FAILED
+Object.freeze(mutationStatus)
+
+/**
+ * Translate a global ID into a token dictionary.
+ *
+ * @param {string} id
+ * @returns {Object}
+ * */
+export function tokenise (id) {
+  if (!id) {
+    return {}
+  }
+  id = id.split('|')
+  const ret = {}
+  for (let ind = 0; ind < id.length; ind++) {
+    ret[identifierOrder[ind]] = id[ind]
+  }
+  return ret
+}
+
+/**
+ * Return the lowest token in the hierarchy.
+ *
+ * @param {Object} tokens
+ * @returns {String}
+ * */
+export function getType (tokens) {
+  let last = null
+  let item = null
+  for (const key of identifierOrder) {
+    item = tokens[key]
+    if (!item) {
+      break
+    }
+    last = key
+  }
+  return last
+}
+
+/**
+ * Convert camel case to words.
+ */
+export function camelToWords (camel) {
+  const result = (camel || '').replace(/([A-Z])/g, ' $1')
+  return result.charAt(0).toUpperCase() + result.slice(1)
+}
+
+/**
+ * Process mutations from an introspection query.
+ *
+ * This adds some computed fields prefixed with an underscore for later use:
+ *   _title:
+ *     Human-readable name for the mutation.
+ *   _icon:
+ *     An SVG path to represent the mutation.
+ *   _shortDescription
+ *     The first line of the mutation description.
+ *   _help
+ *     The remainder of the mutation description.
+ *
+ * @param {object} mutations - Mutations as returned by introspection query.
+ * @param {object} types - Types as returned by introspection query.
+ */
+export function processMutations (mutations, types) {
+  let descLines = null
+  for (const mutation of mutations) {
+    descLines = mutation.description.split(/\n+/)
+    mutation._title = camelToWords(mutation.name)
+    mutation._icon = mutationIcons[mutation.name] || mutationIcons['']
+    mutation._shortDescription = descLines[0]
+    mutation._help = descLines.slice(1).join('\n')
+    processArguments(mutation, types)
+  }
+}
+
+/* Add special fields to mutations args from a GraphQL introspection query. */
+/**
+ * Process mutation arguments from an introspection query.
+ *
+ * This adds some computed fields prefixed with an underscore for later use:
+ *   _title:
+ *     Human-readable name for the mutation.
+ *   _cylcObject:
+ *     The Cylc Object this field relates to if any (e.g. Task).
+ *   _cylcType:
+ *     The underlying GraphQL type that provides this relationship
+ *     (e.g. taskID).
+ *     Note that this type may be buried inside other GraphQL types
+ *     (e.g. NON_NULL, LIST).
+ *   _multiple:
+ *     true if this field accepts multiple values.
+ *   _required:
+ *     true if this field must be provided for the mutation to be called.
+ *
+ * @param {object} mutation - One Mutation as returned by introspection query.
+ * @param {object} types - Types as returned by introspection query.
```

Comment:
Is it `object` or `Array`? Further down we call `      arg._default = getNullValue(arg.type, types)`. In `getNullValue` there's a `for (const type of types) {`… if an `Object` is given, it would—I think—fail with `Uncaught TypeError: types is not iterable` (not sure how to test it from the UI, sorry)